src/idom/server/default.py

```diff
@@ -36,9 +36,15 @@ async def serve_development_app(
 
 
 def use_scope() -> Any:
+    """Return the current ASGI/WSGI scope"""
     return _default_implementation().use_scope()
 
 
+def use_location() -> Location:
+    """Return the current route as a string"""
```

Comment:
So the path will update, it's just that state won't persist after navigating.

Reply:
This behavior isn't going to change in that, if the user truly navigates to a different route on the server, then by doing so, the original WS connection will be closed and then a new one will be opened at the new route which IDOM will then pass down into its context thus "updating" the location information.

Reply:
But keep in mind, that updating the location on-the-fly is going to be required for SPAs, since there won't be a WS reconnect occurring.

Reply:
Definitely. I've been about this and it's doable, and more importantly doable in such a way that we should be able to provide routing via a separate library. The main question is around the API this routing library would use.

Reply:
I think making a general "WS command" framework might be a good idea for extensibility.

Reply:
That sort of exists. Custom JS components can gain access to a [`sendEvent` callback](https://idom-docs.herokuapp.com/docs/guides/escape-hatches/javascript-components.html#custom-javascript-components) that will send arbitrary messages to the server. The messages just need to be of the form `{target: "some-string", data: [an-array, ...]}` where `target` refers to some registered callback on the server.

I'm thinking we can just leverage this same API, but have some conventions around that `target` string.

Reply:
For example, if the `target` were an an empty string. This could, by convention, refer to a global event handler. A message like:

```javascript
{
    target: "",
    data: [ { updatePath: "the/new/path" } ],
}
```

Reply:
Relying on an empty string target is a bit risky, since that could potentially come about from a bug.

I'd do something more explicit like `target: "--command"`

Reply:
We can probably stick with the python convention of `__<name>__`

Reply:
Yep, sounds good!